### PR TITLE
user_registration

### PR DESCRIPTION
**Pull request by pg20**

### CODE REVIEW

File: app/controllers/email_templates_controller.rb [hunk 1]
```diff
@@ -46,11 +46,8 @@ def email_template_params
     params.require(:email_template).permit(:subject, :body, :recipients)
   end
 
-  def search_template_params
```
Comment (DMips) [inline app/controllers/email_templates_controller.rb]: @pg20 why did you change anything here?

File: app/controllers/user_registrations_controller.rb [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+  def update
+
+    @token = params[:id]
+    @user = User.find_by(registration_token: params[:id])
+
+    if @user.blank?
+      not_authenticated
+      return
+    end
+    @user.registration_token = nil
```
Comment (DMips) [inline app/controllers/user_registrations_controller.rb]: instead of setting registration_token and password_confirmation here you can use `tap` in user_params
example:
```ruby
def product_params
  params.require(:product).permit(:name).tap do |whitelisted|
    whitelisted[:data] = params[:product][:data]
  end
end
```

File: app/controllers/email_templates_controller.rb (hunk 1 above)
Comment (pg20) [inline app/controllers/email_templates_controller.rb]: Hi i didnt change anything in this :/ I might have pressed delete button while closing it . Sorry. Can u tell me how to undo it :(

File: db/schema.rb [hunk 3]
```diff
@@ -155,7 +155,11 @@
     t.boolean  "is_missing"
     t.string   "program_country"
     t.string   "timezone"
+<<<<<<< HEAD
```
Comment (DMips) [inline db/schema.rb]: incorrect merge ;)

File: app/controllers/user_registrations_controller.rb [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+
+
+
+  end
+
+  private
+  def user_params
+    params.require(:user).permit(:password).tap do |whitelisted|
+      whitelisted[:password_confirmation] = params[:user][:password_confirmation]
+      whitelisted[:registration_token] = nil
```
Comment (pg20) [inline app/controllers/user_registrations_controller.rb]: check this if it is correct

File: app/controllers/user_registrations_controller.rb [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+    @token = params[:id]
+    @user = User.find_by(registration_token: params[:id])
+
+    if @user.blank?
+      not_authenticated
+      return
+    end
+    if @user.update_attributes(user_params)
+      login(@user.email, user_params[:password])
+      redirect_to(dashboard_mentor_profiles_path, :notice => "You've been succesfully added as Mentor.")
```
Comment (pg20) [inline app/controllers/user_registrations_controller.rb]: I have added automatically login for mentor, anjali will further add condition for mentee. I am closing both my task..Registration and automatic login to dashboard :)

Comment (pg20) [inline app/controllers/user_registrations_controller.rb]: I have added automatically login for mentor, anjali will further add condition for mentee. I am closing both my task..Registration and automatic login to dashboard :)

File: app/controllers/user_registrations_controller.rb (hunk 4 above)
Comment (DMips) [inline app/controllers/user_registrations_controller.rb]: it is :) maybe i would change the name and use something diff than whitelisted but it's correct so I'm going to merge it
